iceoryx_posh/test/moduletests/test_mepoo_mempool.cpp

```diff
@@ -113,10 +113,19 @@ TEST_F(MemPool_test, freeChunk)
         chunks.push_back(reinterpret_cast<uint8_t*>(sut.getChunk()));
     }
 
-    for (uint32_t i = 0; i < NumberOfChunks; ++i)
+    for (uint32_t i = 0; i <= NumberOfChunks; ++i)
```

Comment:
You are testing here two things at once. Could you please leave the current test as it is - but it could use some renaming `AllocatingAndFreeingChunksWorks`

And the test case `EXPECT_DEATH({sut.freeChunk(chunks[i-1]);}, ".*");` should be a double free test case.